src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.Unix.cs

```diff
@@ -31,8 +32,31 @@ internal static string ParseFileName(string path, ZipVersionMadeByPlatform madeB
             }
             else
             {
+                PathInternal.CheckInvalidPathChars(path);
                 return Path.GetFileName(path);
             }
         }
+
+        internal static readonly char[] InvalidPathChars =
```

Comment:
> Add non-partial classes PathInternal_Windows/Unix that contains CheckInvalidChars functions. This wouldn't interfere with regular use of the PathInternal partial class but would allow PathInternal_Windows.CheckInvaidChars to work on Unix.

Yeah, that's along the ideas of what I had in mind.  The goal is basically to have each piece of logic exist in one place such that it can then be included in the various places that need it.  PathInternal.Windows.cs' implementation of CheckInvalidChars would call CheckInvalidChars_Windows, and this project could call the _Windows one directly.